assessmentModel/src/commonMain/kotlin/recorders/MotionRecorderConfiguration.kt

```diff
@@ -0,0 +1,142 @@
+ *
+ * - Raw data:
+ *   - [Accelerometer]: Raw accelerometer reading.
+ *   - [Gyro]: Raw gyroscope reading.
+ *   - [Magnetometer]: Raw magnetometer reading.
+ *
+ * - Calculated values:
+ *   - [Attitude]: Calculated orientation of the device using the gyro and magnetometer (if appropriate).
+ *   - [Gravity]: Calculated vector for the direction of gravity in the coordinates of the device.
+ *   - [MagneticField]: The magnetic field vector with respect to the device for devices with a magnetometer.
+ *   - [RotationRate]: The rotation rate of the device for devices with a gyro.
+ *   - [UserAcceleration]: Calculated vector for the participant's acceleration in the coordinates of the device.
+ *
+ */
+@Serializable
+enum class MotionRecorderType : StringEnum {
+    Accelerometer, Gyro, Magnetometer,
+    Attitude, Gravity, MagneticField, RotationRate, UserAcceleration;
+
+    @Serializer(forClass = MotionRecorderType::class)
+    companion object : KSerializer<MotionRecorderType> {
+        override val descriptor: SerialDescriptor = PrimitiveDescriptor("MotionRecorderType", PrimitiveKind.STRING)
+        override fun deserialize(decoder: Decoder): MotionRecorderType {
+            val name = decoder.decodeString()
+            return values().matching(name) ?: throw SerializationException("Unknown $name for ${descriptor.serialName}. Needs to be one of ${values()}")
+        }
+        override fun serialize(encoder: Encoder, value: MotionRecorderType) {
+            encoder.encodeString(value.name.decapitalize())
+        }
+
+        val raw = setOf(
+            Accelerometer,
+            Gyro,
+            Magnetometer
+        )
+        val calculated = setOf(
+            Attitude,
+            Gravity,
+            MagneticField,
+            RotationRate,
+            UserAcceleration
+        )
+    }
+}
+
+/**
+ * A [MotionRecord] is a serializable implementation of a [SampleRecord] that can be used to record a sample from one
+ * of the motion sensors or calculated vectors. All properties on the record ard defined as `null` to allow using this
+ * same serialization object to define both path markers that are added to the data stream to mark step transitions
+ * and the results of the sensor.
+ *
+ * If the [sensorType] is `null` then this is a path marker. Other values included here are only applicable to certain
+ * sensors. This data structure allows for defining all records within a single file.
+ *
+ * Vector properties:
+ * - [x]: The `x` component of the vector measurement for this sensor sample.
+ * - [y]: The `y` component of the vector measurement for this sensor sample.
+ * - [z]: The `z` component of the vector measurement for this sensor sample.
+ * - [w]: The `w` component of the vector measurement for this sensor sample. (Used by the attitude quaternion.)
+ *
+ * Additional properties:
+ * - [eventAccuracy]: A number marking the sensor accuracy of the magnetic field sensor.
```

Comment:
Should this have a different name? Or does it actually indicate the accuracy of the sample regardless of which sensor and not just for the magnetometer?

Reply:
We have years of code files with the existing name for the field so I'm gonna go with "no".